docs/adr/020-form-validation-library-for-frontend.md

```diff
@@ -0,0 +1,35 @@
+# 20. Form Validation library for frontend
+
+Date: 2023-06-27
+
+## Status
+Accepted
+
+## Context
+We need to implement form validation in our web application. We have yet to evaluate several form validation libraries but have tried one and have decided to use Vest.
+
+## Decision
+We will use Vest as our form validation library. Vest is a lightweight and easy-to-use library that provides a declarative syntax for defining validation rules. It has a small footprint and is well-documented, making it easy for new team members to understand and use.
+
+## Pros
+- Vest is lightweight and easy to use.
+- It provides a declarative syntax for defining validation rules.
+- It has a small footprint and is well-documented.
+- It has a large community of users and contributors, which means that it is actively maintained and supported.
+
+## Cons
+- Vest may not be suitable for complex validation scenarios.
+- It may not have all the features that other form validation libraries offer.
+
+## Consequences
+We predict that using Vest will make it easier and faster to implement form validation in our web application. It will also make it easier for new team members to understand and use our validation code.
+
+We predict that using Vest may not be suitable for complex validation scenarios, but we will evaluate this on a case-by-case basis and consider using other libraries if necessary.
+
+We predict that using Vest will have a positive impact on our development process by reducing the time and effort required to implement form validation.
+
+We predict that using Vest will have a positive impact on the maintainability of our codebase by providing a clear and consistent way to define form validation rules.
+
+## Links
+- [Vest GitHub repository]([Title](https://github.com/ealush/vest))
```

Comment:
The `[Title]` syntax is not necessary and is inconsistent with how we've done links in other markdown files but it's not a problem for me

Reply:
good callout, I'll fix this since i did not mean to keep that in there.